src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Setup/NetCoreRuntimeVersionsRegistryReader.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using Microsoft.Win32;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Setup;
+
+internal sealed class NetCoreRuntimeVersionsRegistryReader
+{
+    // https://github.com/dotnet/designs/blob/96d2ddad13dcb795ff2c5c6a051753363bdfcf7d/accepted/2020/install-locations.md#globally-registered-install-location-new
+
+    private const string RegistryKeyPath = """SOFTWARE\WOW6432Node\dotnet\Setup\InstalledVersions\{0}\sharedfx\Microsoft.NETCore.App""";
+
+    /// <summary>
+    ///     Reads the list of installed .NET Core runtimes for the specified architecture, from the registry.
+    /// </summary>
+    /// <remarks>
+    ///     Returns runtimes installed both as standalone packages, and through VS Setup.
+    ///     Values have the form <c>3.1.32</c>, <c>7.0.11</c>, <c>8.0.0-preview.7.23375.6</c>, <c>8.0.0-rc.1.23419.4</c>.
+    ///     If results could not be determined, <see langword="null"/> is returned.
+    /// </remarks>
+    /// <param name="architecture">The runtime architecture to report results for.</param>
```

Comment:
Please note that this is just x64 or ARM right now.

Reply:
The way I refactored this component is such that it can answer for any architecture. Whether we will ever pass a different architecture is a concern for the calling code. Would you like to see a change on that side?